### ISSUE TITLE
Getting "Uncaught TypeError: Cannot read property 'enable' of undefined" - not sure why...

### ISSUE THREAD

**Issue by MerionLial** [publiclab/Leaflet.DistortableImage#714]
Hi,

first off: thanks for the awesome plugin!

Second: unfortunately I get the "Uncaught TypeError: Cannot read property 'enable' of undefined" error and I'm not sure why. I don't think I'm doing anything special.

**Scope**:
Leaflet map with a few other plugins but nothing that should interfere - marker search, fullscreen mode, responsivePopup, etc.

**Description**:
On my map, I have a few overlays and it would be really cool if they could be dragged and resized and such. What I do is, I get all the information about the overlay images (called detailMaps in the code below as they are higher resolution images of the area they are placed over) from the database, iterate through the resulting array and then load the image overlays accordingly.

It looks like this and works fine with regular overlays. Now when I exchange the regular imageOverlay with distortableImageOverlay and the bounds parameter I'd give the imageOverlay with the corner, I get this in the console:
 ```
leaflet.distortableimage.js:1 Uncaught TypeError: Cannot read property 'enable' of undefined
    at i.<anonymous> (leaflet.distortableimage.js:1)
    at HTMLImageElement.s (leaflet.js:5)
```
The images appear, but all in the map center, not where they should be and the are definitely not editable. Even if the editable option is set to true. My code looks like this:
```
   function addAllDetailMaps(detailMaps){
        detailMaps.forEach(function (item) {
            var detailMapBounds = JSON.parse(decodeURIComponent(item.detail_map_bounds));	//the bounds are stored as an encoded string - here we have to revert it
            createDetailMap(item.detail_map_name, detailMapBounds, item.detail_map_start_zoom, item.detail_map_end_zoom);
        });
    }

    function createDetailMap(name, detailMapBounds, startZoom, endZoom){
        window['detailMap'+name] = L.distortableImageOverlay(mapPath+'detail-maps/'+name+'/'+name+'.png?rand='+simpleRandom(), getCornersFromBounds(detailMapBounds), {
            mode: 'drag',
            editable: false,
            actions: [L.DragAction, L.ScaleAction, L.RevertAction, L.LockAction],
        }).bindPopup(detailMapPopup(name, detailMapBounds, startZoom, endZoom));
            map.addLayer(window['detailMap'+name]); 
    }

    function detailMapPopup(name, detailMapBounds, startZoom, endZoom){
        var detailMapPopup = L.responsivePopup({maxWidth: "auto"})
        content = detailMapForm(name, startZoom, endZoom, detailMapBounds, "update");
        detailMapPopup.setContent(content);
        return detailMapPopup;
    }

    function getCornersFromBounds(bounds){
        var nw = L.latLng(bounds._northEast.lat, bounds._southWest.lng),
            ne = L.latLng(bounds._northEast.lat, bounds._northEast.lng),
            sw = L.latLng(bounds._southWest.lat, bounds._southWest.lng),
            se = L.latLng(bounds._southWest.lat, bounds._northEast.lng);
        var corners= [nw, ne, sw, se];
        return corners;
    }
```

Any help would be appreciated as the functionalities of this plugin look awesome and I would love to use it. 

Note: I'm using this on a "flat image map", meaning xy coords and no real latlng, but it works with regular overlays, so it should with this plugin too, at least I think so.

**Comment by sashadev-sky** [comment]
@MerionLial try:
```
 function createDetailMap(name, detailMapBounds, startZoom, endZoom){
        window['detailMap'+name] = L.distortableImageOverlay(mapPath+'detail-maps/'+name+'/'+name+'.png?rand='+simpleRandom(), {
            corners: getCornersFromBounds(detailMapBounds),
            mode: 'drag',
            editable: false,
            actions: [L.DragAction, L.ScaleAction, L.RevertAction, L.LockAction],
        }).bindPopup(detailMapPopup(name, detailMapBounds, startZoom, endZoom));
            map.addLayer(window['detailMap'+name]); 
    }
```

The corners are meant to be passed in as part of the options hash, not a sep argument.

**Comment by MerionLial** [comment]
Thank you! Should have noticed that, but that helped to at least put the images in the right spot. They are all where they are supposed to be now, But unfortunately I still get the above mentioned error and the images are not clickable (I've set editable: true now for testing purposes). Neither does the toolbar open nor the popup I'm binding myself.

**Comment by sashadev-sky** [comment]
@MerionLial I need to double check about leaflet.draw actually thank you for reminding me!! Also so sorry, the problem is because of documentation not updated.. `L.RevertAction` was updated to `L.RestoreAction` when it was moved from being a WIP. I will fix this in the README accordingly but you need to update that in your code! 

Lmk if all works out, you can let us know if Leaflet.Draw conflict is in fact fixed :) thanks!

**Comment by MerionLial** [comment]
Thanks for the info, I changed my code from `L.RevertAction`  to` L.RestoreAction` , but unfortunately that was not the real problem. I do have a new error for you, though ;-)
I switched browsers and while Chrome now says
`Uncaught TypeError: Cannot read property 'Action' of undefined` 
Firefox is a bit more precise and says 
`Uncaught TypeError: L.Toolbar2 is undefined`
Both now point me to line 922 of the unminified file:
`(L.EditAction = L.Toolbar2.Action.extend({`
which is within this function:
```
    function (t, e) {
        (L.DistortableImage = L.DistortableImage || {}),
            (L.distortableImage = L.DistortableImage),
            (L.DistortableImage.action_map = {}),
            (L.EditAction = L.Toolbar2.Action.extend({
                options: { toolbarIcon: { svg: !1, html: "", className: "", tooltip: "" } },
                initialize: function (t, e, i) {
                    (this._overlay = e), (this._map = t), L.setOptions(this, i), L.Toolbar2.Action.prototype.initialize.call(this, i), this._injectIconSet();
                },
                _createIcon: function (t, e, i) {
                    var o = this,
                        n = this.options.toolbarIcon,
                        a = n.className,
                        s = this._overlay.editing;
                    (this.toolbar = t),
                        (this._icon = L.DomUtil.create("li", "", e)),
                        (this._link = L.DomUtil.create("a", "", this._icon)),
                        n.svg ? (this._link.innerHTML = L.IconUtil.create(n.html)) : (this._link.innerHTML = n.html),
                        this._link.setAttribute("href", "#"),
                        this._link.setAttribute("title", n.tooltip),
                        this._link.setAttribute("role", "button"),
                        L.DomUtil.addClass(this._link, this.constructor.baseClass),
                        a && (L.DomUtil.addClass(this._link, a), "disabled" === a && L.DomUtil.addClass(this._icon, a), a === s._mode ? L.DomUtil.addClass(this._link, "selected-mode") : L.DomUtil.removeClass(this._link, "selected-mode")),
                        L.DomEvent.on(this._link, "click", this.enable, this),
                        L.DomEvent.on(this._overlay, "update", function () {
                            var t = o._link.innerHTML.match(/xlink:href="#restore"/);
                            t && 1 === t.length && o._enableAction();
                        }),
                        this._addSubToolbar(t, this._icon, i);
                },
                _injectIconSet: function () {
                    if (!document.querySelector("#iconset")) {
                        var t = document.createElement("div");
                        (t.id = "iconset"), t.setAttribute("hidden", "hidden"), (t.innerHTML = new L.ToolbarIconSet().render()), document.querySelector(".leaflet-marker-pane").appendChild(t);
                    }
                },
                _enableAction: function () {
                    L.DomUtil.removeClass(this._link.parentElement, "disabled"), L.DomUtil.removeClass(this._link, "disabled");
                },
                _disableAction: function () {
                    L.DomUtil.addClass(this._link.parentElement, "disabled"), L.DomUtil.addClass(this._link, "disabled");
                },
            })),
            (L.editAction = function (t, e, i) {
                return new L.EditAction(t, e, i);
            });
    },
```
 So I guess this might indicate that the conflict with Draw still persists - or that I'm doing something wrong. Is there a common way of resolving this? Should I install this additional plugin? https://github.com/justinmanley/leaflet-draw-toolbar

Sorry if my questions sound a bit stupid, but I'm just getting the hang of leaflet itself but haven't played around with toolbars or tried my hand at adapting plugins as complex as this one.

**Comment by sashadev-sky** [comment]
@MerionLial Not stupid!! I checked the compatibility myself and it worked for me. What versions of everything are you using? I think maybe your version of leaflet-toolbar is old. You should have `"leaflet-toolbar": "0.4.0-alpha.2",`

**Comment by MerionLial** [comment]
Well, this is a bit embarrassing, but I did not have `leaflet-toolbar` included in my list of loaded scripts. Changed that and inlcuded `leaflet-toolbar` and it seems to work now.
I was not aware that this is a prerequisite of `Leaflet.DistortableImage`. I simply downloaded the zip and added the .js to my plugin folder. When you use npm, such dependencies get automatically included, I guess. 
Not sure if there are many others nowadays that do it 'manually' like me, but maybe adding a note to the read.me could be helpful?
Anyway, thank you for your efforts which were successful in the end!
